### PR TITLE
Added auth context

### PR DESCRIPTION
**Pull request by surajeaton**
Fixes # BLUI-5117

#### Changes proposed in this Pull Request:

- Added auth context, provider, hook
- Added test cases
- Added example

#### Screenshots / Screen Recording (if applicable)

- 
Simulator Screenshot - iPhone 12 - 2024-03-13 at 16 03 40

#### To Test:

- cd login-workflow
- yarn start:example
- Added console to see email in `forgotPassword` in AuthUIActions
- Open drawer
- Press Auth Provider
- Check the console

#### Any specific feedback you are looking for?

-

### CODE REVIEW

File: login-workflow/src/contexts/AuthContext/provider.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import { ErrorContext } from '../ErrorContext';
+import { AuthDictionaries } from './AuthDictionaries';
+import { SharedDictionaries } from '../SharedDictionaries';
+import { ErrorManagerProps } from '../../components/Error';
+
+const AuthContextProviderContent: React.FC<
+    React.PropsWithChildren<Omit<AuthContextProviderProps, 'i18n'> & { PasswordDialog?: JSX.Element }>
+> = (props) => {
+    const { children, errorConfig, ...authContextProps } = props;
+    const mergedErrorConfig: ErrorManagerProps = {
```
Comment (JeffGreiner-eaton) [inline login-workflow/src/contexts/AuthContext/provider.tsx:20]: I notice the useTranslation and t function missing for mergedErrorConfig but we have translation strings in here.

**Comment by JeffGreiner-eaton** [comment]
I see console in the auth custom screen, do we need to add console for email?
At the moment just running the example and selecting authProvider from drawer opens auth screen but no logs and screen stuck here..

image
image

File: login-workflow/src/contexts/AuthContext/provider.tsx (hunk 1 above)
Comment (surajeaton) [reply login-workflow/src/contexts/AuthContext/provider.tsx:21]: Updated the type

**Comment by surajeaton** [comment]
> I see console in the auth custom screen, do we need to add console for email? At the moment just running the example and selecting authProvider from drawer opens auth screen but no logs and screen stuck here..
> 
> image image

That's right, you should be able to see screen and when you press on Next, there should be console

File: login-workflow/src/contexts/AuthContext/context.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/**
+ * @packageDocumentation
+ * @module AuthContext
+ */
+import { createContext } from 'react';
+import { AuthContextProviderProps } from './types';
+
+export const AuthContext = createContext<AuthContextProviderProps | null>(null);
```
Comment (ektaghag-eaton) [inline login-workflow/src/contexts/AuthContext/context.ts:8]: Add one line of description in jsdoc similar to this https://github.com/etn-ccis/blui-react-native-workflows/blob/release/R31/login-workflow/src/contexts/RegistrationContext/context.ts#L10

File: login-workflow/src/contexts/AuthContext/types.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
+/**
+ * @packageDocumentation
+ * @module AuthContext
+ */
+
+import { i18n } from 'i18next';
+import { RouteConfig } from '../../types';
+import { ErrorContextProviderProps } from '../ErrorContext/types';
+
+export type AuthUIActions = {
```
Comment (ektaghag-eaton) [inline login-workflow/src/contexts/AuthContext/types.ts:10]: please recheck jsdocs with registrationContext for ref. https://github.com/etn-ccis/blui-react-native-workflows/blob/release/R31/login-workflow/src/contexts/RegistrationContext/types.ts#L16

File: login-workflow/src/contexts/AuthContext/provider.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import { AuthContextProviderProps } from './types';
+import { AuthContext } from './context';
+import { I18nextProvider, useTranslation } from 'react-i18next';
+import { i18nAuthInstance } from './i18nAuthInstance';
+import { ErrorContext } from '../ErrorContext';
+import { AuthDictionaries } from './AuthDictionaries';
+import { SharedDictionaries } from '../SharedDictionaries';
+import { ErrorManagerProps } from '../../components/Error';
+
+const AuthContextProviderContent: React.FC<
```
Comment (ektaghag-eaton) [inline login-workflow/src/contexts/AuthContext/provider.tsx:16]: Add jsdoc

**Comment by ektaghag-eaton** [comment]
AuthUIActions are working fine. Fix above PR comments.
Screenshot 2024-03-15 at 11 45 29 AM

**Pull request merged**